parsers/jscript.c

```diff
@@ -745,7 +745,10 @@ static void skipArgumentList (tokenInfo *const token, boolean include_newlines,
 		while (nest_level > 0 && ! isType (token, TOKEN_EOF))
 		{
 			readTokenFull (token, FALSE, repr);
-			if (isType (token, TOKEN_OPEN_PAREN))
+			if (isType (token, TOKEN_KEYWORD) && token->keyword == KEYWORD_function &&
+			    repr == NULL)
+				parseFunction (token);
+			else if (isType (token, TOKEN_OPEN_PAREN))
```

Comment:
you could avoid changing this line by adding an `else if` to the end of the chain rather than prepending it.   It wouldn't change anything as the other types handled are different from `KEYWORD` anyway.